src/lib.rs

```diff
@@ -729,7 +741,11 @@ pub fn shutdown_logger_raw() -> Result<*const Log, ShutdownLoggerError> {
         // FIXME add a sleep here when it doesn't involve timers
     }
 
-    Ok(unsafe { LOGGER.unwrap() })
+    unsafe {
+        let logger = LOGGER;
+        LOGGER = &NopLogger;
```

Comment:
Reset `LOGGER` here for slightly better hygiene and to avoid providing a permanent root for the pointer that might make Valgrind miss a memory leak.